tools/gyp/pylib/gyp/generator/make.py

```diff
@@ -1758,8 +1758,15 @@ def WriteMakeRule(self, outputs, inputs, actions=None, comment=None,
       self.WriteLn('%s: %s' % (' '.join(outputs), intermediate))
       self.WriteLn('\t%s' % '@:')
       self.WriteLn('%s: %s' % ('.INTERMEDIATE', intermediate))
-      self.WriteLn('%s: %s%s' %
-                   (intermediate, ' '.join(inputs), force_append))
+      # Don't add `force_append` (FORCE_DO_CMD) to the intermediate sentinal.
+      # Adding it makes the action run alway, even when there are no changes.
+      # Excluding macOS which has a different way of resolving dependancies
+      # which can lead to deadlocks.
+      # (refack): AFAICT because `*.intermediate` files don't have build rules.
+      if self.flavor == 'mac':
+        self.WriteLn('%s: %s%s' % (intermediate, ' '.join(inputs), force_append))
+      else:
+        self.WriteLn('%s: %s' % (intermediate, ' '.join(inputs)))
```

Comment:
This is the wrong way of doing it. You're effectively ignoring the value of the `force` argument to `WriteMakeRule()` here.

The call comes from `WriteDoCmd()`. Forcing the action to run when it's a command is correct; commands are imperatives. You should try to come up with another way of solving this.

Reply:
Yeah, I know this is a hack. But the soup GYP generates for `make` is hard to reason about, and empirically this works (it also works on macOS, but might lead to deadlocks when run `-jN` with N > 1).

I'll give it more thought...